### DISCUSSION TITLE
Auth working in localhost, but not working when deployed to cloud run

### DISCUSSION THREAD

**Discussion by dmetoki** [vercel/next.js#74329] [Help] [answered]
### Summary

Hi all,
I have implemented authentication in Next js 15, using Authjs v5, and Google as the provider.
Everything works fine locally, but when I deploy the application in Production, and click on the sign in button, I get an error 500.
I've checked the 'Authorized redirect URIs' and they look good. I have set my environment variables both, with the 'NEXT', prefix and declaring them explicitly, but it made no difference. I have also included the env variables in the service config in cloud run. So, everything seems to be set correctly, but it seems that something is blocking the communication between my app and the provider.
Has anyone experienced something like this? Is this a default config in cloud run that I should be editing?
Below the key parts of my codebase. Any help would be much appreciated.

**api/auth/[...nextauth].route.tsx**
```
import { handlers } from "@/auth"; // Referring to the auth.ts we just created
export const { GET, POST } = handlers

```

**auth.tsx**
```
import NextAuth from "next-auth";
import Google from "next-auth/providers/google";
 
export const { handlers, signIn, signOut, auth } = NextAuth({
  providers: [Google],
})
```

**middleware.tsx**
`export { auth as middleware } from "@/auth";`

**app/page.tsx**
```
import SignIn from "@/app/components/sign-in";
import { SignOut } from "@/app/components/sign-out";
import { auth } from "@/auth";

export default async function Home() {
  const session = await auth()
  console.log("Session:", session); // Debugging session value
  
  return (
    <div className="grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]">
      <main className="flex flex-col gap-8 row-start-2 items-center sm:items-start">
        <div>
          {session ? <SignOut /> : <SignIn />}
        </div>
      </main>
    </div>
  );
}
```
**app/components/sign-in.tsx**
```
import { signIn } from "@/auth"
export default function SignIn() {
  return (
    <form
      action={async () => {
        "use server"
        await signIn("google")
      }}
    >
      <button type="submit" className="rounded-full border border-solid border-transparent transition-colors flex items-center justify-center bg-foreground text-background gap-2 hover:bg-[#383838] dark:hover:bg-[#ccc] text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5">Signin with Google</button>
    </form>
  )
} 
```

**app/components/sign-in.tsx**
```
import { signOut } from "@/auth"
export function SignOut() {
  return (
    <form
      action={async () => {
        "use server"
        await signOut()
      }}
    >
      <button type="submit" className="rounded-full border border-solid border-black/[.08] dark:border-white/[.145] transition-colors flex items-center justify-center hover:bg-[#f2f2f2] dark:hover:bg-[#1a1a1a] hover:border-transparent text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5 sm:min-w-44">Sign Out</button>
    </form>
  )
}
```

Thanks in advance for your help

**Comment by icyJoseph** [comment]
> Everything works fine locally,

Does that include `next build && next start` locally?

**Reply by icyJoseph** [reply, marked as answer]
Have you also gotten a change to add more debugging within your `signIn` function? Make sure you have a proper try/catch, so that you can see at which step it fails.

**Reply by dmetoki** [reply]
Thanks @icyJoseph 
yup. next build and next start are working fine locally.
I'll try adding more logging. The problem is that the logs will be captured through google cloud run logging, and unfortunately they don't provide any info other than showing that the call to the api fails with an error 500

**Reply by icyJoseph** [reply]
🤔 but if you wrap it on a proper try catch, perhaps you can avoid the server error altogether, and get back a reason, for failure, on your server action? if no reason for failure comes downstream, then it was successful, __sort of__

**Reply by dmetoki** [reply]
I finally figured out what was wrong.
My AUTH_URL was set to the plain version of the domain https://mywebsite.com
When accessing the website through https://www.mywebsite.com, google authentication failed.
I'm not sure how to force the redirection so that i consolidate everything under one domain, or setting two different AUTH URLs but at least the mystery is solved.
Thanks for your advice @icyJoseph . The additional logging helped.
